docs/spec/consensus/fork-accountability.md

```diff
@@ -0,0 +1,142 @@
+all it's data for the current unbonding period to a correct node that will inspect this data and submit corresponding
+evidence. There are also architectures that assumes a special role (sometimes called fisherman) whose goal is to collect
+as much as possible useful data from the network, to do analysis and create evidence transactions. That functionality
+is outside the scope of this document.
+
+Note that after lite client is fooled by a fork, that normally means that an attacker can change in an arbitrarily
+way application state and validator set. Difference between LCS and LCB might only be in the amount of voting power
+needed to convince lite client about arbitrary state. In case of LCB where security threshold is at minimum (+1/3 of 
+voting power of trusted validator set), an attacker can arbitrarily modify application state with +1/3 of voting power,
+while in case of LCS it requires +2/3 of voting power. 
+
+### Amnesia based attacks
+
+In case of amnesia faulty validator lock some value v in some round r, and then vote for different value v' in higher
+rounds without correctly unlocking value v. This attack can be used both on full nodes and lite clients. 
+
+Example scenario: +2/3 of voting power commit a value A in round r. Remaining validators haven't locked any value.
++1/3 of faulty validators ignore the fact that they locked a value A and propose a different block B in the following
+rounds r' > r. As 1/3 of correct processes haven't locked any value they would accept proposal for B and they commit
+a block B. Note that in this case +1/3 of faulty validators does not need to commit an equivocation as they only vote once per round. Detecting faulty validators in case of such attack require fork accountability mechanism described in 
+this document https://docs.google.com/document/d/11ZhMsCj3y7zIZz4udO9l25xqb0kl7gmWqNpGVRzOeyY/edit?usp=sharing. 
+
+If a lite client is attacked using this attack with +1/3 of voting power, attacker can not arbitrarily change application state. In case there is an attack with +2/3 of voting power, an attacker can arbitrarily change application state. Note that in this case at least +1/3 of faulty validators locked value A in round r, so if they sign different value in follow up rounds this will be detectable by the the fork accountability mechanisms. The remaining 1/3 of faulty processes might not locked value A in round r so they can't be detected using this mechanism. Only in case they signed something which conflicts with the application this can be used against them. Otherwise they don't do anything incorrect. Note that this case is not covered by the report https://docs.google.com/document/d/11ZhMsCj3y7zIZz4udO9l25xqb0kl7gmWqNpGVRzOeyY/edit?usp=sharing as it only assumes at most 2/3 of faulty validators.
+
+Q: do we need to define a special kind of attack for the case where a validator sign arbitrarily state? It seems that
```

Comment:
Can we simply slash bonded or unbonding validators for signing (with their consensus keys) *anything* except the actual application state committed on the canonical chain - or are you talking about a case where the application state could have been valid (if it was signed in a different round from the canonical commit)?